File: unity/Assets/Scripts/unityagent/Management/ManagementMessageManager.cs
using UnityAgent.Buffers;
using UnityAgent.SystemMessage;

namespace UnityAgent.Management.Manager
{
public abstract class ManagementMessageManager
{
    public abstract void ProcessIsm(IngressSystemMessage ism);

    protected void SendEsm(EgressSystemMessage esm)
    {
        EgressSystemMessageBuffer esmb = new();
        esmb.Enqueue(esm);
        EgressBuffer.Enqueue(esmb);
    }
}
}

File: ros/ros_packages/rosagent/rosagent/management/management_message_manager.py
from rosagent.buffers import egress_buffer as EgressBuffer
from rosagent.logging.rosagent_logger import RosagentLogger as Logger
from rosagent.message.egress_system_message_buffer import EgressSystemMessageBuffer

from rosutility.message.serializer import deserializer as Deserializer

class ManagementMessageManager:
    
    def __init__(self):
        self._logger = Logger()

    def _send_esm(self, esm):
        esm_buffer = EgressSystemMessageBuffer(size=1)
        esm_buffer.enqueue(esm)
        EgressBuffer.enqueue(esm_buffer)

    def _deserialize_management_message_data(self, ism):
        data = ism.get_data()
        management_data = Deserializer.bytes_to_dict(
            data, create_snake_case_keys=True)
        return management_data
